tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,41 +1,291 @@
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+
+  expect_equal(
+    expected,
+    filter(actual, PARAMCD %in% c("WTASDS", "WTAPCTL")) %>% pull(AVAL)
+  )
+})
+
+## Test 2: Height SDS and percentile works (P50, P97) ----
+test_that("derive_params_growth_age Test 2: Height SDS and percentile works (P50, P97)", {
+  #LL: LENGTH?
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "1001", "M", 2, "months", "HEIGHT", 58,
+    "1002", "F", 578, "days", "HEIGHT", 87,
+    "1003", "M", 910, "days", "HEIGHT", 92,
+    "1004", "F", 159, "months", "HEIGHT", 170,
+  )
+
+  # pull WHO metadata for patients <2 years old, CDC metadata for patients >=2 years old
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 1, 58.4384, 0.03423,
+    "F", 578, "days", 1, 81.6752, 0.03619,
+    "M", 910, "days", 0.2418532658, 90.91041609, 0.0409575918,
+    "F", 4840, "days",1.172254081, 158.2058471, 0.04304528616,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "HEIGHT",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "HTASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "HTAPCTL"
     )
   )
 
-  expected <- c(
-    ((10 / 2)^1 - 1) / (1 * 3),
-    ((100 / 8)^7 - 1) / (7 * 9)
+  vs_data_age <- vs_data %>%
+    mutate(AGECUR_c = case_when(
+      AGEU == "days"  ~ AGECUR,
+      AGEU == "months" ~ round(AGECUR*30.4375)),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"),-AGECUR)
+
+  vs_data_meta <- vs_data_age %>%inner_join(meta, by=c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>%mutate(AVAL = (((VSSTRESN/M)^L)-1)/(L*S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL)*100)
+  expected <- bind_rows(vs_sds, vs_pctl)  %>% pull(AVAL)
+
+
+  expect_equal(
+    expected,
+    filter(actual, PARAMCD %in% c("HTASDS", "HTAPCTL")) %>% pull(AVAL)
+  )
+})
+
+## Test 3: BMI SDS and percentile works (Z-score of -2, 0, 2, 5) ----
+test_that("derive_params_growth_age Test 3: BMI SDS and percentile works (Z-score of -2, 0, 2, 5)", {
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "1001", "M", 2, "months", "BMI", 58,
+    "1002", "F", 578, "days", "BMI", 87,
+    "1003", "M", 2146, "days", "BMI", 13.5,
+    "1004", "F", 88.5, "months", "BMI", 21.9,
+  )
+
+  # pull WHO metadata for patients <2 years old, CDC metadata for patients >=2 years old
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 0.1113, 16.3231, 0.08676,
+    "F", 578, "days", -0.5197, 15.6524, 0.08631,
+    "M", 2146, "days", -3.15039004, 15.37745304, 0.081706249,
+    "F", 2694, "days",-2.819657704, 15.56444426, 0.109308364,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
```

Comment:
Should the new argument `by_vars` be added there ?

Reply:
Thanks for catching! Yes, `by_vars` added to all test codes.